code/game/objects/structures/lavaland/geyser.dm

```diff
@@ -65,6 +65,8 @@
 		start_chemming()
 
 /obj/structure/geyser/attackby(obj/item/item, mob/user, params)
+	. = ..()
+
 	if(!istype(item, /obj/item/mining_scanner) && !istype(item, /obj/item/t_scanner/adv_mining_scanner))
 		return
```

Comment:
If I remember right, attackby is one of the few procs that you sometimes shouldn't call parent in and causes you to hit the target. Did you check if attacking with the scanner actualy hits the geyzer. You'll probably want this to be `return ..()` instead of parent call first thing.